### PR TITLE
New Workflow: Lookup VM Host Info Teams

### PR DESCRIPTION
**Pull request by sengleson-r7**

### CODE REVIEW

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/help.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Once the workflow has been imported, **edit the Settings step to specify the Microsoft Teams team name and channel name to suit your Teams environment!** Edit the input with the preset text of `Your Team` and `Your Channel` in each Teams step in the workflow.
+
+After configuring the Settings step with your Microsoft Teams details, activate the workflow in order to trigger it.
+
+## Usage
+
+*This workflow will only trigger in the channel specified in the Microsoft Teams workflow steps.*
+
+To run the workflow, send a message to the specified Microsoft Teams channel starting with the command `!insert-command-here`.
```
Comment (jschipp-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/help.md:33]: Given they we're shipping a specific command in this workflow I don't see a reason to be generic about this here.

```suggestion
To run the workflow, send a message to the specified Microsoft Teams channel starting with the command `!insert-command-here`.
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/workflow.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+name: Lookup_InsightVM_Host_Info_From_Teams
+title: "Lookup InsightVM Host Info from Microsoft Teams"
+description: Lookup a host, including IP, MAC, last assessment date, vulnerability statistics, and running services, from a Slack message.
+version: 1.0.1
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [vulnerability_management, asset_inventory]
+  keywords: [InsightVM, Slack, chat, work_from_home]
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [InsightVM, chatops, chat, work_from_home]
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/workflow.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+extension: workflow
+products: ["insightconnect", "insightvm"]
+name: Lookup_InsightVM_Host_Info_From_Teams
+title: "Lookup InsightVM Host Info from Microsoft Teams"
+description: Lookup a host, including IP, MAC, last assessment date, vulnerability statistics, and running services, from a Slack message.
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/workflow.spec.yaml:5]:
```suggestion
description: Lookup a host, including IP, MAC, last assessment date, vulnerability statistics, and running services, from a Microsoft Teams message.
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/help.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+# Version History
+
+* 1.0.0 - Initial workflow
+
+# Links
+
+## References
+
+* [Rapid7 InsightVM](https://www.rapid7.com/products/insightvm)
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/help.md:65]:
```suggestion
* [Rapid7 InsightVM](https://www.rapid7.com/products/insightvm)
* [Microsoft Teams](https://teams.microsoft.com)
* [Microsoft Teams Setup](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/Lookup_InsightVM_Host_Info_From_Teams.icon [hunk 5]
```diff
@@ -0,0 +1,2181 @@
...
+          },
+          "1a20436a-d71c-4732-8b36-0dc6ad7012b4": {
+            "nodeId": "1a20436a-d71c-4732-8b36-0dc6ad7012b4",
+            "name": "Final Report",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "# Trigger\n_{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[user]}}_ started Get InsightVM Host Info workflow with the search term _{{[61855db7-9d9e-46ab-8176-6e18cd3d8299].[argument]}}_ in the _{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[channel]}}_ channel at _{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[timestamp]}}_.\n\n# Search Results\n_{{[61855db7-9d9e-46ab-8176-6e18cd3d8299].[argument]}}_ yielded the following search results in InsightVM:\n{{# each [d79b7dd3-bc74-4649-8e24-67680740b43e].[assets]}}\n{{this.[hostName]}}\n{{/each}}\n\n# Outcome\nPosted asset details to the _{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[channel]}}_ channel in Slack."
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/Lookup_InsightVM_Host_Info_From_Teams.icon:253]:
```suggestion
                "content": "# Trigger\n_{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[user]}}_ started Get InsightVM Host Info workflow with the search term _{{[61855db7-9d9e-46ab-8176-6e18cd3d8299].[argument]}}_ in the _{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[channel]}}_ channel at _{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[timestamp]}}_.\n\n# Search Results\n_{{[61855db7-9d9e-46ab-8176-6e18cd3d8299].[argument]}}_ yielded the following search results in InsightVM:\n{{# each [d79b7dd3-bc74-4649-8e24-67680740b43e].[assets]}}\n{{this.[hostName]}}\n{{/each}}\n\n# Outcome\nPosted asset details to the _{{[8b5f067d-03aa-4d9e-a06b-5ad7dda77b99].[message].[channel]}}_ channel in Microsoft Teams."
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/Lookup_InsightVM_Host_Info_From_Teams.icon [hunk 6]
```diff
@@ -0,0 +1,2181 @@
...
+  "kom": {
+    "komandVersion": "1.51.0-0-ge64bd",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-05-27T17:12:40.543009265Z",
+    "workflowVersions": [
+      {
+        "name": "Lookup Host in InsightVM from Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "This workflow accepts a Teams command containing a hostname or IP address and returns asset details from InsightVM.\n\nSample Slack Trigger Commands: \n!lookup-host workstation123\n!lookup-host 192.168.1.2",
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/Lookup_InsightVM_Host_Info_From_Teams.icon:11]:
```suggestion
        "description": "This workflow accepts a Teams command containing a hostname or IP address and returns asset details from InsightVM.\n\nSample Teams Trigger Commands: \n!lookup-host workstation123\n!lookup-host 192.168.1.2",
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/Lookup_InsightVM_Host_Info_From_Teams.icon [hunk 7]
```diff
@@ -0,0 +1,2181 @@
...
+              "slugName": "type_converter",
+              "slugVersion": "1.5.1",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/type_converter/1.5.1/icon.png"
+            },
+            "identifier": "string_to_object",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "input": "{\"team\":\"Your Team\",\"channel\":\"Your Channel\"}"
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/Lookup_InsightVM_Host_Info_From_Teams.icon:279]:
```suggestion
                "input": "{\"team\":\"change_me\",\"channel\":\"change_me\"}"
```

File: workflows/Lookup_InsightVM_Host_Info_From_Teams/help.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+* InsightConnect License
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Once the workflow has been imported, **edit the Settings step to specify the Microsoft Teams team name and channel name to suit your Teams environment!** Edit the input with the preset text of `Your Team` and `Your Channel` in each Teams step in the workflow.
```
Comment (zyoutz-r7) [inline workflows/Lookup_InsightVM_Host_Info_From_Teams/help.md:25]:
```suggestion
Once the workflow has been imported, **edit the Settings step to specify the Microsoft Teams team name and channel name to suit your Teams environment!** Edit the input with the preset text of `change_me` in each Teams step in the workflow.
```

**Pull request merged**